agent/03_agent_deployment.sh

```diff
@@ -198,15 +198,47 @@ EOF
     set -x
 }
 
-function generate_fleeting_iso() {
-    export REPO_PATH=${WORKING_DIR}
+function sync_installer() {
+    local DEST="${OPENSHIFT_INSTALL_PATH}"
+    if [ ! -d "${DEST}" ]; then
+        echo "Syncing $DEST"
+
+        mkdir -p "${DEST}"
+        git clone https://github.com/openshift/installer.git "${DEST}"
+
+        pushd "${DEST}"
 
-    sync_repo_and_patch fleeting https://github.com/openshift-agent-team/fleeting ${FLEETING_PR}
+        DEFAULT_BRANCH=agent-installer
 
-    generate_fleeting_manifests
+        git am --abort || true
+        git checkout "${DEFAULT_BRANCH}"
```

Comment:
I'm undecided about the right thing to do here. Currently it only checks out the branch the first time, upon cloning.
The `sync_repo_and_patch` code always checks out the branch and rebases it on the upstream. So that would mean you can't test a change locally on a branch other than `agent-installer`.
It seems to me that you sometimes want to automatically rebase, but sometimes don't. I'm not sure how to determine which is which.

Reply:
The would appear to combine the worst of both worlds - you can't work on any local branch except `agent-installer`, and you have to manually update it.

Reply:
When working on a local branch the test on  `OPENSHIFT_INSTALL_PATH` should avoid (re)checking out the branch or the PR (as it happens in the build_installer() function, even though the current function does not manage the additional PR param) - since the `sync_repo_and_patch` should be required only when a local checkout is not available - thus allowing the build installer to build whatever it's found in the `OPENSHIFT_INSTALL_PATH` - while here I just noticed that (if I'm not wrong with the tabs alignment) the PR checkout is always executed and maybe this caused some confusion (at least for me).

Considering also that for the CI we'll not need to manage the code at all (but the installer will have to be extracted from the ephemeral release payload), it could be worth evaluating to reuse in some way the whole [03_build_installer.sh](https://github.com/openshift-metal3/dev-scripts/blob/master/03_build_installer.sh) as it is by adjusting a couple of integration points (namely, the checkout of the `agent-installer` branch instead of the `main`, and probably the `TAGS` when building it). In this way we could continue to use the already existing `KNI_INSTALL_FROM_GIT` feature when developing locallly (with the possibility to override `OPENSHIFT_INSTALL_PATH` when required).